plugins/discourse-reactions/assets/javascripts/discourse/models/discourse-reactions-custom-reaction.js

```diff
@@ -7,12 +7,12 @@ import Topic from "discourse/models/topic";
 import User from "discourse/models/user";
 
 export default class CustomReaction extends RestModel {
-  static toggle(post, reactionId) {
+  static toggle(post, reactionId, appEvents) {
     return ajax(
       `/discourse-reactions/posts/${post.id}/custom-reactions/${reactionId}/toggle.json`,
       { type: "PUT" }
     ).then((result) => {
-      post.appEvents.trigger("discourse-reactions:reaction-toggled", {
+      appEvents.trigger("discourse-reactions:reaction-toggled", {
```

Comment:
Since this can now be our `op_reactions_data` or a real post, we can't access `appEvents` on `op_reactions_data`... so we pass it in :)